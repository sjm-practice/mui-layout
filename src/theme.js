import { createMuiTheme } from "@material-ui/core/styles";
import { red } from "@material-ui/core/colors";

// Create a theme instance.
const theme = createMuiTheme({
  palette: {
    error: {
      main: red.A400,
    },
  },
});

export default theme;
