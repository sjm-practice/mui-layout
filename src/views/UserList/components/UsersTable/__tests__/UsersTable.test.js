/* eslint-env jest */

import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "@material-ui/styles";
import theme from "../../../../../theme";
import userData from "../../../data";

import UsersTable from "../index";

describe("<UsersTable />", () => {
  it("Should match Snapshot", () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <UsersTable users={userData} />
      </ThemeProvider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
