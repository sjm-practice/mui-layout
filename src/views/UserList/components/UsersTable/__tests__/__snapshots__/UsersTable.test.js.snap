// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersTable /> Should match Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-root-1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root makeStyles-content-2"
    >
      <div
        class="scrollbar-container ps"
      >
        <div
          class="makeStyles-inner-3"
        >
          <table
            class="MuiTable-root"
          >
            <thead
              class="MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox"
                  scope="col"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-60 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                      <input
                        class="PrivateSwitchBase-input-63"
                        data-indeterminate="false"
                        type="checkbox"
                        value=""
                      />
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Name
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Email
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Location
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Phone
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Registration date
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-hover"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-60 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                      <input
                        class="PrivateSwitchBase-input-63"
                        data-indeterminate="false"
                        type="checkbox"
                        value="true"
                      />
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  <div
                    class="makeStyles-nameContainer-4"
                  >
                    <div
                      class="MuiAvatar-root makeStyles-avatar-5"
                    >
                      <img
                        class="MuiAvatar-img"
                        src="/static/images/avatars/avatar_3.png"
                      />
                    </div>
                    <p
                      class="MuiTypography-root MuiTypography-body1"
                    >
                      Erik Tankova
                    </p>
                  </div>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  erik.tankova@devias.io
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  Parkersburg, West Virginia, USA
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  304-428-3097
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  11/04/2019
                </td>
              </tr>
              <tr
                class="MuiTableRow-root MuiTableRow-hover"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-60 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                      <input
                        class="PrivateSwitchBase-input-63"
                        data-indeterminate="false"
                        type="checkbox"
                        value="true"
                      />
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  <div
                    class="makeStyles-nameContainer-4"
                  >
                    <div
                      class="MuiAvatar-root makeStyles-avatar-5"
                    >
                      <img
                        class="MuiAvatar-img"
                        src="/static/images/avatars/avatar_4.png"
                      />
                    </div>
                    <p
                      class="MuiTypography-root MuiTypography-body1"
                    >
                      Cao Yu
                    </p>
                  </div>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  cao.yu@devias.io
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  Iowa, Bristow, USA
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  712-351-5711
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  11/04/2019
                </td>
              </tr>
              <tr
                class="MuiTableRow-root MuiTableRow-hover"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-60 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                      <input
                        class="PrivateSwitchBase-input-63"
                        data-indeterminate="false"
                        type="checkbox"
                        value="true"
                      />
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  <div
                    class="makeStyles-nameContainer-4"
                  >
                    <div
                      class="MuiAvatar-root makeStyles-avatar-5"
                    >
                      <img
                        class="MuiAvatar-img"
                        src="/static/images/avatars/avatar_2.png"
                      />
                    </div>
                    <p
                      class="MuiTypography-root MuiTypography-body1"
                    >
                      Alexa Richardson
                    </p>
                  </div>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  alexa.richardson@devias.io
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  Atlanta, Georgia, USA
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  770-635-2682
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  11/04/2019
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="ps__rail-x"
          style="left: 0px; top: 0px;"
        >
          <div
            class="ps__thumb-x"
            style="left: 0px; width: 0px;"
            tabindex="0"
          />
        </div>
        <div
          class="ps__rail-y"
          style="top: 0px; left: 0px;"
        >
          <div
            class="ps__thumb-y"
            style="top: 0px; height: 0px;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiCardActions-root makeStyles-actions-6 MuiCardActions-spacing"
    >
      <div
        class="MuiTablePagination-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
        >
          <div
            class="MuiTablePagination-spacer"
          />
          <span
            class="MuiTypography-root MuiTablePagination-caption MuiTypography-caption MuiTypography-colorInherit"
          >
            Rows per page:
          </span>
          <div
            class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
          >
            <div
              aria-haspopup="listbox"
              class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input MuiInputBase-inputSelect"
              role="button"
              tabindex="0"
            >
              10
            </div>
            <input
              type="hidden"
              value="10"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <span
            class="MuiTypography-root MuiTablePagination-caption MuiTypography-caption MuiTypography-colorInherit"
          >
            1-3 of 3
          </span>
          <div
            class="MuiTablePagination-actions"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </span>
            </button>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
